### ISSUE TITLE
Inference results of fine-tuned LayoutLM models differ depending on input ID and box order

### ISSUE THREAD

**Issue by rahelhabacker** [microsoft/unilm#1098]
Hi, I have fine-tuned LayoutLM (v1) on my own invoice data. The model, after 4 epochs, reaches a pretty good performance.

<img width="1018" alt="Screenshot 2023-05-23 at 15 14 57" src="https://github.com/microsoft/unilm/assets/8261902/f3d78342-b3ed-4cbe-90cd-27eabc57877d">

When using it for inference, though, I get different outputs depending on the order of `input_ids` and `bbox` tensors in the encoding. The difference I observe is mostly based on the order of `other` words versus words with any label except `other`. There is three different orderings I have tested:

1. first all words/boxes semantic interesting boxes (i.e. boxes are expected to be classified with any label except `other`), then all `other` boxes
2. random order of labeled `other` versus non-`other` words/boxes
3. words/boxes ordered by the box position, top left to bottom right

When I run the inference, I get different results as follows:

1. First non-other boxes/words, then other boxes/words
<img width="500" alt="ner_output_1" src="https://github.com/microsoft/unilm/assets/8261902/19a5d233-df63-419a-99a9-6df481f75ac2">

2. Random order
<img width="500" alt="ner_output_2" src="https://github.com/microsoft/unilm/assets/8261902/eefc22c8-e74c-4490-b6df-5fdd03ff6aac">

3. Top left to bottom right order
<img width="500" alt="ner_output_3" src="https://github.com/microsoft/unilm/assets/8261902/938f1e73-69ac-48b3-96b5-aa07a2e21f0d">

Case 1 matches the ground truth the most... this is the word/box order with which the model also has been trained. However, the difference in results between the cases is not what I expected... I expect the same results for all these cases, i.e. that the result is independent of how words/boxes in the encoding for inference are ordered. 

If the word/box order is relevant, what is the correct order for training and for inference? Or is the only important thing that the inference order matches the training order?

If useful, I can provide the encoding and fine-tuned model.

**Comment by suresh1505** [comment]
Have you worked on object detection(LayoutLMv3) ? I am not able to get the  input_ids ,bbox  and attention_masks tensors in the encoding.
